openquake/commonlib/sourceconverter.py

```diff
@@ -90,21 +91,30 @@ def area_to_point_sources(area_src, area_src_disc):
 
 def split_fault_source(src):
     """
-    Generator splitting a fault source into several fault sources,
-    one for each magnitude.
+    Generator splitting a fault source into several fault sources.
 
     :param src:
         an instance of :class:`openquake.hazardlib.source.base.SeismicSource`
     """
-    i = 0  # split source index
+    # NB: the splitting is tricky; if you don't split, you will not
+    # take advantage of the multiple cores; if you split too much,
+    # the data transfer will kill you, i.e. multiprocessing/celery
+    # will fail to transmit to the workers the generated sources.
+    # Heere I (MS) have set MAX_RUPTURE_SPLITTING to 100, meaning that
+    # at worse the data transfer will increase by a factor of 100 and at
+    # most 100 cores will be used for a source of large enough magnitude
     max_mag = src.get_min_max_mag()[1]
-    if max_mag > MAGNITUDE_FOR_RUPTURE_SPLITTING:
-        for rupture in src.iter_ruptures():
-            i += 1
+    if (max_mag > MAGNITUDE_FOR_RUPTURE_SPLITTING and
+            src.count_ruptures() <= MAX_RUPTURE_SPLITTING):
+        # split on SingleRuptureSources
+        for i, rupture in enumerate(src.iter_ruptures()):
             yield SingleRuptureSource(
                 rupture, '%s-%s' % (src.source_id, i),
                 src.tectonic_region_type, src.trt_model_id)
-    else:  # split on the annual occurrence rates
+    else:
+        # too small magnitude or too many ruptures:
+        # split on annual occurrence rates
```

Comment:
How do you know this? Are you using this function only when one of the two conditions is met i.e. when the number of ruptures generated by a single source would be too large?